contracts/ModuleFactory.sol

```diff
@@ -0,0 +1,44 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+import "./modules/RevokeModule.sol";
+import "./modules/AaveWithdrawModule.sol";
+
+import "interfaces/gnosis/IGnosisSafe.sol";
+
+contract ModuleFactory {
+    ////////////////////////////////////////////////////////////////////////////
+    // ERRORS
+    ////////////////////////////////////////////////////////////////////////////
+
+    error NoSupportedModuleType(ModuleType moduleType, address deployer);
+
+    error NotSigner(address safe, address executor);
+
+    ////////////////////////////////////////////////////////////////////////////
+    // EVENTS
+    ////////////////////////////////////////////////////////////////////////////
+
+    event ModuleDeployed(address module, ModuleType moduleType, address deployer, uint256 timestamp);
+
+    ////////////////////////////////////////////////////////////////////////////
+    // CONSTANTS
+    ////////////////////////////////////////////////////////////////////////////
+
+    enum ModuleType {
+        REVOKE_MODULE,
+        AAVE_WITHDRAW
+    }
+
+    function createModuleAndEnable(IGnosisSafe safe, ModuleType modType) external {
+        if (modType == ModuleType.REVOKE_MODULE) {
+            RevokeModule module = new RevokeModule();
+            emit ModuleDeployed(address(module), ModuleType.REVOKE_MODULE, msg.sender, block.timestamp);
+        } else if (modType == ModuleType.AAVE_WITHDRAW) {
+            AaveWithdrawModule module = new AaveWithdrawModule();
+            emit ModuleDeployed(address(module), ModuleType.AAVE_WITHDRAW, msg.sender, block.timestamp);
```

Comment:
i think technically these two emits could be reduced to one on L43: `emit ModuleDeployed(address(module), modType, msg.sender, block.timestamp);`

Reply:
module is not avail outside of the if-else block